### PR TITLE
interfaces,overlord,testutil: implement support for interface-based security

### PR DESCRIPTION
**Pull request by zyga**
This branch adds 2nd iteration of "security helpers" that know enough to enable certain type of security support (apparmor, seccomp, udev, dbus) for a given snap. Unlinke the first version they have somewhat different layout a public API (through interfaces.SecurityConfigurator) and are meant to be used from the overlord directly.

The branch has extensive tests for apparmor, basic tests for other backends and no tests for the new interface manager task (I'm mostly looking for feedback at that level and I probably will only land that after the snap manager populates some basic facts about snaps).

### CODE REVIEW

File: interfaces/apparmor/cfg.go [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+		return fmt.Errorf("cannot obtain security snippets for snap %q: %s", snapInfo.Name, err)
+	}
+	// Get the files that this snap should have
+	dir, glob, content, err := cfg.DirStateForInstalledSnap(snapInfo, developerMode, snippets)
+	if err != nil {
+		return fmt.Errorf("cannot obtain expected apparmor files for snap %q: %s", snapInfo.Name, err)
+	}
+	// Ensure that files are correctly on disk
+	changed, removed, err := osutil.EnsureDirState(dir, glob, content)
+	// Record changes so that we can do clean-up in Finalize. This has to
```
Comment (pedronis) [inline interfaces/apparmor/cfg.go]: All ConfigureSnapSecurity seem to start the same? first impression is that the abstraction is not the best if it ends up with that much repetition? what's the relation between Configurators and Security Systems?

File: interfaces/udev/cfg.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	if len(removed) > 0 {
+		cfg.needsReload = true
+	}
+	if err != nil {
+		return fmt.Errorf("cannot synchronize udev files for snap %q: %s", snapInfo.Name, err)
+	}
+	return nil
+}
+
+// Finalize does nothing at all.
```
Comment (pedronis) [inline interfaces/udev/cfg.go]: doesn't seem to do nothing here?

Comment (zyga) [inline interfaces/udev/cfg.go]: Ah, copy paste error. I'll correct this.

**Comment by zyga** [comment]
Closing for the split

**Pull request closed without merging**